src/neptune_scale/core/metadata_splitter.py

```diff
@@ -166,3 +173,22 @@ def populate_tags(
                 break
 
         return size
+
+    def _skip_non_finite(self, step: Optional[float | int], metrics: Dict[str, float]) -> Iterator[Tuple[str, float]]:
+        """Yields (metric, value) pairs, skipping non-finite values depending on the env setting."""
+
+        for k, v in metrics.items():
+            v = float(v)
+
+            if not math.isfinite(v):
+                if SKIP_NON_FINITE_METRICS:
```

Comment:
This is because the env is evaluated on import, so the flag retains the initial value. I'll change this, it's not a big effort and could help with interactive shells